src/Core/Bases/A11yElement.cs

```diff
@@ -417,7 +419,7 @@ public T GetPlatformPropertyValue<T>(int propertyId)
         {
             var property = this.PlatformProperties?.ById(propertyId);
             if (property == null) return default;
-            if (!(property.Value is T)) throw new AxeWindowsException(Invariant($"Expected property.Value, which is type {property.Value.GetType().Name}, to be type {typeof(T).Name}"));
+            if (!(property.Value is T)) throw new AxeWindowsException(String.Format(CultureInfo.InvariantCulture, ErrorMessages.PropertyValueTypeUnexpected, property.Value.GetType().Name, typeof(T).Name));
```

Comment:
A potentially localizable string + CultureInfo.InvariantCulture seems a bit strange. Part of me wants to use CultureInfo.CurrentCulture here, so that we're all set if/when we localize exception messages, although that might make the exception messages a little harder to process. Another part of me wants to have some sort of extension method that hides the formatting in a single place, so that this line might look something like this:
```
if (!(property.Value is T)) throw new AxeWindowsException(ErrorMessages.PropertyValueTypeUnexpected.With2Parameters(property.Value.GetType().Name, typeof(T).Name));
```

The With2Parameters method would own the formatting, the CultureInfo, and perhaps some way to call out errors if the string and the parameter count somehow get out of sync.

To be 100% clear, these are just musings--feel free to ignore them as appropriate

Reply:
The usage of CultureInfo.InvariantCulture all over the place was bothering me too, so I think the extension method is a good idea. I've implemented it in the latest commit, let me know what you think!